ray_data_eval/image_training/sampling.py

```diff
@@ -22,27 +36,59 @@ def copy_random_folders(source_dir, target_dir, percentage=0.05, copy_dir=True):
     # returns num_to_copy *unique* elements, so no repetition
     selected_dirs = random.sample(subdirs, num_to_copy)
 
-    for subdir in selected_dirs:
-        src_path = os.path.join(source_dir, subdir)
-        dst_path = os.path.join(target_dir, subdir)
-        if copy_dir:
-            shutil.copytree(src_path, dst_path)
-        else:
-            shutil.copy2(src_path, dst_path)
-        print(f"Copied {src_path} to {dst_path}")
+    # init multiprocessing pool
+    pool = multiprocessing.Pool(processes=multiprocessing.cpu_count())
```

Comment:
Good! I personally find the API of [`ProcessPoolExecutor`](https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.ProcessPoolExecutor) a little easier to use. But for this simple enough task, either will suffice.